src/graphics/Screen.h

```diff
@@ -82,6 +82,14 @@ class Screen
 #define SEGMENT_WIDTH 16
 #define SEGMENT_HEIGHT 4
 
+static uint16_t displayWidth, displayHeight;
+
+#define SCREEN_WIDTH displayWidth
+#define SCREEN_HEIGHT displayHeight
```

Comment:
Just from some initial investigation, the waypoint compass rose issue comes from how these are accessed. I'll take a look at it in the next few hours unless you're already working on it!